lib/Installation/Warnings/WarningsController.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The class introduces business actions for Warning Popups
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+package Installation::Warnings::WarningsController;
+use strict;
+use warnings;
+
+use strict;
+use warnings;
+use YuiRestClient;
+use Installation::Warnings::ConfirmationWarningRichText;
+
+sub new {
+    my ($class, $args) = @_;
+    my $self = bless {}, $class;
+    return $self->init();
+}
+
+sub init {
+    my ($self) = @_;
+    $self->{ConfirmationWarningRichText} = Installation::Warnings::ConfirmationWarningRichText->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub get_confirmation_warning_rich_text {
```

Comment:
the caller might not care about if it is rich text, or you plan to use in future `get_confirmation_warning_text` ? Could be just get_text(), but it will be answer above about the reason to have this controller.

Reply:
Agree. I used the approach taken from another place in our framework, but as we identify better approaches constantly, I would say, having just `get_text()` is the proper way.

In the past it was not so clear on how it will affect us while having different warning popups, so we used this approach with separating them. Now we identified we should go towards unifying things that are same from the business logic of product. So, warning is warning, client shouldn't care about it's internals.